### PR TITLE
Added type definitions for jsstore

### PR DESCRIPTION
**Pull request by ujjwalguptaofficial**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by plantain-00 — CHANGES_REQUESTED** [review]
The library is not on npm, since you are the library's owner, you can publish it to npm with types, that will be better.
If you really want to publish your types here, you missed `tslint.json` file.

File: types/jsstore/tslint.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "defaultSeverity": "error",
+    "extends": [
+        "tslint:recommended"
```
Comment (plantain-00) [inline types/jsstore/tslint.json]: Use `{ "extends": "dtslint/dt.json" }` rather than `tslint:recommended`

**Comment by ujjwalguptaofficial** [comment]
@plantain-00 - I have added the library on npm. Here is the link - https://www.npmjs.com/package/jsstore

please review again and let me know if there is any other requirement or improvements needed to done. Fyi - I can not add jslint, because when extending dtslint/dt.json , other rules are not working and the library has some code which needs to override some rules.

**Comment by plantain-00** [comment]
If the package bundles types, you don't have to create types in this repository, because this repository is for the packages that can not bundle `.d.ts` file, so you can close this PR, users can consume the types from the package `jsstore` rather than `@types/jsstore` see: https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/declaration%20files/Publishing.md
On the other hand, you didn't set the path of type file in `package.json` file, it should be something like: `"types": "./Dist/Latest/V 1.1.9/JsStore-1.1.9.d.ts"`  see: https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/declaration%20files/Publishing.md#including-declarations-in-your-npm-package
And usually, packages do not need to publish all of the versions, if users want to use `v1.1.9`, they can use it by `npm install jsstore@1.1.9`; if users want to use `v1.1.8`, they can use it by `npm install jsstore@1.1.8`. it is what package's `version` is used for.

File: types/jsstore/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+    module Business {
+        module Update {
+            var updateValue: (suppliedValue: any, storedValue: any) => any;
+        }
+        class BaseUpdate extends Base {
+            CheckFlag: boolean;
+        }
+    }
+}
+declare module JsStore {
```
Comment (RyanCavanaugh) [inline types/jsstore/index.d.ts]: All of these re-openings of the module should be changed to a single declaration. This is adding a lot of noise for no reason - the file should be ~40% smaller than it is rightn ow

File: types/jsstore/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,705 @@
...
+import DataBase = Model.DataBase;
+import Column = Model.Column;
+import Table = Model.Table;
+declare module JsStore {
+    class Instance extends CodeExecutionHelper {
+        constructor(dbName?: any);
+        createDb(dataBase: Model.IDataBase, onSuccess?: Function, onError?: Function): this;
+        dropDb(onSuccess: Function, onError?: Function): this;
+        select(query: ISelect, onSuccess?: Function, onError?: Function): this;
+        count(query: ICount, onSuccess?: Function, onError?: Function): this;
```
Comment (RyanCavanaugh) [inline types/jsstore/index.d.ts]: All these `Function`s should be more specific types

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

**Comment by RyanCavanaugh** [comment]
> - I can not add jslint, because when extending dtslint/dt.json , other rules are not working and the library has some code which needs to override some rules.

This needs more justification. You can suppress any individual warning with a suppression comment

**Comment by ujjwalguptaofficial** [comment]
Thanks for your help guys and I figured it out that I can do this directly from npm as @plantain-00 said. So I am closing this request.

**Pull request closed without merging**